libbeat/monitoring/report/elasticsearch/elasticsearch.go

```diff
@@ -52,66 +48,93 @@ var defaultParams = map[string]string{
 }
 
 func init() {
-	report.RegisterReporterFactory("elasticsearch", New)
+	report.RegisterReporterFactory("elasticsearch", makeReporter)
 }
 
-func New(beat common.BeatInfo, cfg *common.Config) (report.Reporter, error) {
+func makeReporter(beat common.BeatInfo, cfg *common.Config) (report.Reporter, error) {
 	config := defaultConfig
 	if err := cfg.Unpack(&config); err != nil {
 		return nil, err
 	}
 
-	clientFactory, err := makeClientFactory(&config)
+	// check endpoint availablity on startup only every 30 seconds
+	checkRetry := 30 * time.Second
+	windowSize := config.BulkMaxSize - 1
+	if windowSize <= 0 {
+		windowSize = 1
+	}
+
+	proxyURL, err := parseProxyURL(config.ProxyURL)
+	if err != nil {
+		return nil, err
+	}
+	if proxyURL != nil {
+		logp.Info("Using proxy URL: %s", proxyURL)
+	}
+	tlsConfig, err := outputs.LoadTLSConfig(config.TLS)
 	if err != nil {
 		return nil, err
 	}
 
-	clients, err := modeutil.MakeClients(cfg, clientFactory)
+	params := map[string]string{}
+	for k, v := range config.Params {
+		params[k] = v
+	}
+	for k, v := range defaultParams {
+		params[k] = v
+	}
+	params["interval"] = config.Period.String()
+
+	out := outputs.Group{
+		Clients:   nil,
+		BatchSize: windowSize,
+		Retry:     0, // no retry. on error drop events
+	}
+
+	hosts, err := outputs.ReadHostList(cfg)
 	if err != nil {
 		return nil, err
 	}
+	for _, host := range hosts {
+		client, err := makeClient(host, params, proxyURL, tlsConfig, &config)
+		if err != nil {
+			return nil, err
+		}
+		out.Clients = append(out.Clients, client)
+	}
 
-	// backoff parameters
```

Comment:
This is the monitoring reporter, not the actual Elasticsearch event output.
The backoff parameters have become configurable, as one has(will have to) configure the backoff strategy when creating the `outputs.Group`. This is due this PR shifting some responsibilities from the outputs to the publisher pipeline itself (e.g. error handling, backoff, retry). Some of the shifting will be necessary, when introducing dynamic output reloading, as the pipeline must transfer active events from old to new outputs.